### PR TITLE
Add explicit info about # of inventory findings and rule findings

### LINKED ISSUE #320: "Rule Recommendation" results included in non-blocking findings count
When using the "Rule Recommendation" functionality of Semgrep App, the CI runner displays very confusing messages like `2 current issues found` and `2 non-blocking findings hidden`.

### PR DESCRIPTION
**Pull request by raghavjain3**
Closes https://github.com/returntocorp/semgrep-action/issues/320

### CODE REVIEW

File: src/semgrep_agent/semgrep.py [hunk 1]
```diff
@@ -187,9 +187,21 @@ def get_findings(
...
                         click.echo(f"|    {s}", err=True)
+            inventory_findings_len = 0
+            for finding in findings.current:
+                if "__r2c-internal-cai" in finding.check_id:
+                    inventory_findings_len += 1
             click.echo(
                 f"| {unit_len(findings.current, 'current issue')} found", err=True
             )
+            click.echo(
+                f"| {unit_len(range(len(findings.current)-inventory_findings_len), 'current rule issue')}",
```
Comment (brendongo) [inline src/semgrep_agent/semgrep.py:198]: Why is range necessary?

File: src/semgrep_agent/semgrep.py [hunk 2]
```diff
@@ -187,9 +187,21 @@ def get_findings(
                 for e in findings.errors:
                     for s in render_error(e):
                         click.echo(f"|    {s}", err=True)
+            inventory_findings_len = 0
+            for finding in findings.current:
```
Comment (brendongo) [inline src/semgrep_agent/semgrep.py:191]: Might be useful to factor into a function to count cai findings given list of findings

File: src/semgrep_agent/semgrep.py (hunk 1 above)
Comment (raghavjain3) [reply src/semgrep_agent/semgrep.py:198]: unit_len requires an iterable as a parameter so unit_len(int) doesn't work but unit_len(range(int)) does

**Review by brendongo — APPROVED** [review]

File: tests/acceptance/baseline-local-file/new-agent.err [hunk 3]
```diff
@@ -10,6 +10,8 @@
 | found 2 files in the paths to be scanned
 === looking for current issues in 2 files
 | 1 current issue found
+| 1 current rule issue
```
Comment (dlukeomalley) [inline tests/acceptance/baseline-local-file/new-agent.err:13]: What is current rule issue vs. a current issue?

File: tests/acceptance/baseline-local-file/new-agent.err [hunk 4]
```diff
@@ -10,6 +10,8 @@
...
+| No current inventory issues
```
Comment (dlukeomalley) [inline tests/acceptance/baseline-local-file/new-agent.err:14]: What are inventory issues? HMW make it clear issue doesn't imply "problem" in this case, like it's not something for a user to go fix?

I wonder if we just signal that rule recommendation rules ran, and avoid any specific counts that confuse users or they think they've specifically configured.

Comment (chmccreery) [reply tests/acceptance/baseline-local-file/new-agent.err:14]: Big +1 from me

Comment (raghavjain3) [reply tests/acceptance/baseline-local-file/new-agent.err:14]: Will create a PR today to fix this :)

**Pull request merged**